### PR TITLE
WEB-299: Web Content API Client Endpoints

### PR DESCRIPTION
**Pull request by lwilkowskeBC**
Functionality for Web Content API. Ability to get, create, update and delete pages under the "Storefront Content > Web Pages" menu in the CP

@aleachjr @lord2800 @bc-joshroe @ICarpenter

### CODE REVIEW

**Comment by ICarpenter** [comment]
👍

**Comment by lord2800** [comment]
No tests?

**Comment by lwilkowskeBC** [comment]
@lord2800 ♻️

**Comment by lord2800** [comment]
How about some tests for the `Client` methods? You should be able to use the existing `Client` tests as a guide.

**Comment by lwilkowskeBC** [comment]
@lord2800 ♻️

File: test/Unit/Api/ClientTest.php [hunk 1]
```diff
@@ -322,6 +323,11 @@ public function testGettingTheCountOfACollectionReturnsThatCollectionsCount($pat
             return;
         }
 
+        //This isn't really something needed ATM for this group of functions
+        if ($path == 'pages'){
```
Comment (lord2800) [inline test/Unit/Api/ClientTest.php]: You should move this to the `in_array` check above.

**Comment by lord2800** [comment]
Other than that, this is awesome. Nice work!

File: test/Unit/Api/ClientTest.php (hunk 1 above)
Comment (lwilkowskeBC) [inline test/Unit/Api/ClientTest.php]: Oh duh. For some reason I was not reading that as `in_array` but some imaginary function that was checking for all params to exist in string.

**Comment by lord2800** [comment]
Much 👍.

**Pull request merged**